### PR TITLE
8345178: RISC-V: Add gtests for narrow cmpxchg

### PR DESCRIPTION
**Pull request by robehn**
Hi, please consider.

This adds tests of aligned narrow cmpxchg.

```
Note: Google Test filter = *RiscV*
[==========] Running 5 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 5 tests from RiscV
...
[ RUN      ] RiscV.cmpxchg_int16_lr_sc_vm
[       OK ] RiscV.cmpxchg_int16_lr_sc_vm (2 ms)
[ RUN      ] RiscV.cmpxchg_int8_lr_sc_vm
[       OK ] RiscV.cmpxchg_int8_lr_sc_vm (1 ms)
[ RUN      ] RiscV.cmpxchg_int16_maybe_zacas_vm
[       OK ] RiscV.cmpxchg_int16_maybe_zacas_vm (1 ms)
[ RUN      ] RiscV.cmpxchg_int8_maybe_zacas_vm
[       OK ] RiscV.cmpxchg_int8_maybe_zacas_vm (1 ms)
[----------] 5 tests from RiscV (20831 ms total)

[----------] Global test environment tear-down
[==========] 5 tests from 1 test suite ran. (20834 ms total)
[  PASSED  ] 5 tests.
```

Executed with -XX:+UnlockExperimentalVMOptions -XX:+UseZacas

Thanks, Robbin

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8345178](https://bugs.openjdk.org/browse/JDK-8345178): RISC-V: Add gtests for narrow cmpxchg (**Sub-task** - P4)

### Reviewers
 * [Fei Yang](https://openjdk.org/census#fyang) (@RealFYang - **Reviewer**)
 * [Hamlin Li](https://openjdk.org/census#mli) (@Hamlin-Li - **Reviewer**) Review applies to [22e8c4e6](https://git.openjdk.org/jdk/pull/22445/files/22e8c4e61d488d79e7c1d4289fb34508c1eacca5)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22445/head:pull/22445` \
`$ git checkout pull/22445`

Update a local copy of the PR: \
`$ git checkout pull/22445` \
`$ git pull https://git.openjdk.org/jdk.git pull/22445/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22445`

View PR using the GUI difftool: \
`$ git pr show -t 22445`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22445.diff">https://git.openjdk.org/jdk/pull/22445.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22445#issuecomment-2506474251)
</details>

### CODE REVIEW

**Review by RealFYang — APPROVED** [review]
Thanks.

**Comment by robehn** [comment]
> Thanks for adding the tests!
> 
> The test code looks good. Just some minor comment below.
> 
> And one general question, will we consider to add some concurrent tests in the future? Although I don't have a good idea how to implement such tests in a stable way. :)

Thanks, yes I'm adding concurrent tests also.
There is several in the gtest suit.

File: test/hotspot/gtest/riscv/test_assembler_riscv.cpp [hunk 1]
```diff
@@ -106,4 +105,90 @@ TEST_VM(RiscV, cmov) {
...
+void run_narrow_cmpxchg_tests() {
+  // Assume natural aligned
+  TESTSIZE data[8];
+  TESTSIZE ret;
+  for (int i = 0; i < 7; i++) {
+    memset(data, -1, sizeof(data));
+
+    data[i] = 121;
+    ret = NarrowCmpxchgTester<TESTSIZE, ASMSIZE>::narrow_cmpxchg((intptr_t)&data[i], 121, 42, /* result */ 67,
+                                                                          -1, -1, -1, false);
```
[the remark this answers is not in the document]
Comment (robehn) [reply test/hotspot/gtest/riscv/test_assembler_riscv.cpp:144]: Thanks!

**Review by Hamlin-Li — APPROVED** [review]
Thanks for updating, looks good!

**Review by RealFYang — APPROVED** [review]

**Pull request closed without merging**